zerver/views/realm_filters.py

```diff
@@ -41,3 +41,24 @@ def delete_filter(request: HttpRequest, user_profile: UserProfile, filter_id: in
     except RealmFilter.DoesNotExist:
         return json_error(_("Filter not found"))
     return json_success()
+
+
+@require_realm_admin
+@has_request_variables
+def update_filter(
+    request: HttpRequest,
+    user_profile: UserProfile,
+    filter_id: int,
+    pattern: str = REQ(),
+    url_format_string: str = REQ(),
+) -> HttpResponse:
+    try:
+        do_update_realm_filter(
+            realm=user_profile.realm,
+            id=filter_id,
+            pattern=pattern,
+            url_format_string=url_format_string,
+        )
+        return json_success()
+    except ValidationError as e:
+        return json_error(e.messages[0], data={"errors": dict(e)})
```

Comment:
This needs to catch RealmFilter.DoesNotExist.  You should use the same error message as for the deletion code path above.

Reply:
But updating an existing filter is like adding a filter, hence errors each time can be different (cannot be always `RealmFilter.DoesNotExist`) like `Invalid filter pattern` (when new filter pattern entered is Invalid) or `Invalid URL format string` (when the new format string is Invalid).
Should I still update it to `RealmFilter.DoesNotExist` ?
Or should I add this as extra except block to catch `RealmFilter.DoesNotExist` ?

Reply:
Yeah, I meant an additional except block -- sorry for the confusion.  I agree we still need the `except ValidationError` block that you already have.